gensim/models/wrappers/fasttext.py

```diff
@@ -256,7 +257,14 @@ def load_binary_data(self, model_binary_file):
             self.load_vectors(f)
 
     def load_model_params(self, file_handle):
-        (dim, ws, epoch, minCount, neg, _, loss, model, bucket, minn, maxn, _, t) = self.struct_unpack(file_handle, '@12i1d')
+        magic, v= self.struct_unpack(file_handle, '@2i')
+        if magic == 793712314:  # newer format 
+            self.new_format = True
+            dim, ws, epoch, minCount, neg, _, loss, model, bucket, minn, maxn, _, t = self.struct_unpack(file_handle, '@12i1d')
+        else:  # older format
```

Comment:
Better to set `self.new_format = False` inside this `else`, rather than inside `initialize_word_vectors`